addons/binding/org.openhab.binding.elkm1/src/main/java/org/openhab/binding/elkm1/discovery/ElkM1DiscoveryHandler.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2019 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.elkm1.discovery;
+
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.elkm1.ElkM1BindingConstants;
+import org.openhab.binding.elkm1.handler.ElkM1BridgeHandler;
+import org.openhab.binding.elkm1.internal.ElkM1HandlerListener;
+import org.openhab.binding.elkm1.internal.elk.ElkTypeToRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Sets up the discovery results and details from the elk m1 when it is found.
+ *
+ * @author David Bennett - Initial Contribution
+ */
+public class ElkM1DiscoveryHandler extends AbstractDiscoveryService implements ElkM1HandlerListener {
+    private Logger logger = LoggerFactory.getLogger(ElkM1DiscoveryHandler.class);
+
+    private ElkM1BridgeHandler bridge;
+
+    public ElkM1DiscoveryHandler(ElkM1BridgeHandler bridge) throws IllegalArgumentException {
+        super(60);
+        this.bridge = bridge;
+    }
+
+    @Override
+    public void activate(Map<String, Object> configProperties) {
+        logger.info("Elk M1 Binding Activated");
```

Comment:
Please check the log level of all logging statements. This logging (and most other logging) should be done on debug. See the [Logging Guidelines](https://www.openhab.org/docs/developer/development/guidelines.html#e-logging).